### ISSUE TITLE
Widescan does not differentiate between maps

### ISSUE THREAD

**Issue by Wiggo32** [DarkstarProject/darkstar#4600]
**_I have:_**

- [x] searched existing issues (http://github.com/darkstarproject/darkstar/issues/) to see if the issue I am posting has already been addressed or opened by another contributor
- [x] checked the commit log to see if my issue has been resolved since my server was last updated

**_Client Version_** (type `/ver` in game) **:** 30171223_0

**_Source Branch_** (master/stable) **:** fae4c89

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 
1. Go anywhere that there are different levels corresponding to separate maps for a zone.
    (I've found Phomiuna Aqueducts to be a good place to see what's happening here)
2. Use widescan to locate mobs and note that you can see mobs on different maps than the one you are currently on.

**Comment by TeoTwawki** [comment]
I've been able to see mobs on diff map floors with widescan on retail so i don't think it checks which map I am on but I am pretty sure we should have a different vertical range limit instead of treating it as a sphere of the widescan value and that would be missing in DSP.

**Comment by Rihyuh** [comment]
That would be a mesh layer issue? Sorta like when skeletons would come up out of the ground if you fought cemetery cherry and blood aggro went off ^.^

**Comment by TeoTwawki** [comment]
no, unless someonewants to prove thats how retail does it (checking if there is a floor/cieling between you and every mob in the zone before deciding to show it on the widescan list or not  is highly unlikely).


we probably just need to limit the vertical range - right now your widescan is omnidirectional

**Comment by Rihyuh** [comment]
Oh word. Didn't realize widescan acted like a sphere in the mechanics. I
had thought it was a distance circumference on a predetermined size based
on job, RNG and BST being larger, that used the mesh layers to separate
level breaks.
Be neat to have a program that would show the volume around the sprite.

**Comment by Wiggo32** [comment]
So, I've been running around Vanadiel for the past two weeks trying to find mobs that I can see on my map, that 'aren't on my map'... and I think I need a little guidance on where to look. So far the only mobs I can see on my map are the ones that are actually on my map according to where they are standing vs what map I would show if I stood in the same spot. I even went to oztroja in a spot where the maps change and had a yagudo oracle standing 5 yalms away from me and not be on my widescan map, although it's elemental was... =) The closest thing I could get to having mobs not be on my map was when I would see mobs which looked "off the map" because that part of the map is missing so it looked like the mobs were possibly on a different map but in all actuality were still on the map I was on. You can see an example of this at 1:55 on this vid: https://www.youtube.com/watch?v=b_NkFaKp_NE
So, other than that, I need some suggestions for where to look in order to find this widescan occurrence which gets me mobs from a different map on my widescan.

**Comment by TeoTwawki** [comment]
> So, other than that, I need some suggestions for where to look in order to find this widescan occurrence which gets me mobs from a different map on my widescan

palb. mines usually has floors and ceilings pretty close together in a lot of places. You know the places where if no navmesh, you can agro a turtle that comes down from the ceiling? see if on retail you can find that turtle on widescan. It'd look a lot like it was there with you on widescan but in the room you'd see no turtle because he's on the floor above. That room above you is aprox 5 yalms off I think. If you can't see any turtles on widescan that you can't see on screen outside of ws in that same general area I think that's a good indication something might that height is involved.

we can probably packet cap whats in the widescan list and compare to mob_spawn_points too.

**Comment by Wiggo32** [comment]
k, I'll take a run through palborough and capture everything

**Comment by Wiggo32** [comment]
So, I'm convinced that widescan handles which mobs appear on your current map the same way that the map handles which party members appear on your current map. The entity is most likely flagged to be on a 'certain' map based on their position in the zone. And whatever procedure that entails to make sure you are viewing the correct map when you check your map (it has to tell you which map (1, 2, 3, etc.) to look at somehow) that should be the same procedure for every entity be it mob, player, or npc. Here's the video and captures for the ENTIRE Palborough Mines zone with special emphasis on the places where the maps change from one to the next. Never found a mob from a different map on my widescan in all of the two weeks since this issue was opened. Anyone have any other ideas? This is my best guess.
https://www.youtube.com/watch?v=pmvsVPfO3YQ&feature=youtu.be

**Comment by TeoTwawki** [comment]
well I'd say safe bet you ruled out a simple height check. truly baffling.
